projects/plugins/jetpack/extensions/blocks/ai-paragraph/attributes.js

```diff
@@ -7,4 +7,8 @@ export default {
 		type: 'boolean',
 		default: false,
 	},
+	wasNotCompletedWhenSaved: {
```

Comment:
I don't like the single-purpose attribute here that will live with this syntax forever.

This is also why originally I had an attribute storing essentially the same information: That the prompt has been already requested:
This error is exactly the reason why I originally had the attribute that the prompt has been requested:

https://github.com/Automattic/jetpack/pull/28043/files#diff-91c04cc3a869b9e81137a4b77a523ee8e773f029e3af26153b5383feb397f8a4R6


I propose a similar attribute.
Or better yet - what about storing the entire prompt if it was sent? This will be future-proof and pair nicely with the entire work of https://github.com/Automattic/jetpack/issues/29310

Reply:
> I don't like the single-purpose attribute here that will live with this syntax forever.
> 
> This is also why originally I had an attribute storing essentially the same information

I don't understand this feedback: you don't like it, and yet you had the same thing originally?

> I propose a similar attribute.

Isn't that a similar attribute already?

>  what about storing the entire prompt if it was sent

Yes, I like that idea! However having the prompt sent does not mean the request was completed - that was also the issue with `requestedPrompt`: if there is an error in the request, the block will again disappear on page save/reload...
I like the idea of storing the prompt, but we will have to keep `wasNotCompletedWhenSaved`

Reply:
> I don't understand this feedback: you don't like it, and yet you had the same thing originally?

It seems to me that `requestedPrompt` can be potentially more useful in the future compared to `wasNotCompletedWhenSaved`.

What I am proposing is a little reflection before exploding the attribute list with a separate attribute for every single regression we have.

> Yes, I like that idea! However having the prompt sent does not mean the request was completed - that was also the issue with requestedPrompt: if there is an error in the request, the block will again disappear on page save/reload...

Request was not completed when `content` is empty. `wasNotCompletedWhenSaved` is an equivalent to `( prompt && ! content )` unless I am missing something

Reply:
If we can have `content` and `prompt` as attributes seems also more elegant because these are essentially the inputs that went into producing this block: prompt from the user and content from openai

Reply:
> Request was not completed when content is empty. wasNotCompletedWhenSaved is an equivalent to ( prompt && ! content ) unless I am missing something

I see the issue! The problem is after ANY save, `content` is not longer empty. It has a paragraph in it and potentially other data... It tried. to document it [here](https://github.com/Automattic/jetpack/pull/29384/files#diff-c1c872a5561d9c663212cb8aa374644d8589b6b2489de7e27d90f5fcf347f70aR200)

Reply:
> If we can have `content` and `prompt` as attributes seems also more elegant because these are essentially the inputs that went into producing this block: prompt from the user and content from openai

Same comment. I actually lost one hour trying to understand why the code was not working, because of what was in `content` 😡